opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -361,6 +363,8 @@ def end(self):
             has_ended = self.end_time is not None
             if not has_ended:
                 self.end_time = util.time_ns()
+            if end_time is not None:
```

Comment:
If `end_time` is specified, we don't want to call `util.time_ns()` and then drop it on the floor?